srv/components/controller/files/scripts/provision.sh

```diff
@@ -1010,6 +1023,55 @@ ftp_enable()
     } && echo "ftp_enable(): ftp is already enabled" >> $logfile
 }
 
+# Check if sftp service is enabled on controller
+is_sftp_enabled()
+{
+    _tmp_file="$tmpdir/is_sftp_enabled"
+    [ -f $_tmp_file ] && rm -rf $_tmp_file
+    # objects name in the xml
+    _xml_obj_bt="security-communications-protocols"
+    _xml_obj_plist=("sftp")
+    # run command to get the details of advanced settings params
+    echo "is_sftp_enabled():Getting protocols details.." >> $logfile
+    _cmd="show protocols"
+    cmd_run "$_cmd"
+    echo "is_sftp_enabled():Checking if sftp service is enabled" >> $logfile
+
+    # parse xml to get required values of properties
+    parse_xml $xml_doc $_xml_obj_bt "${_xml_obj_plist[@]}" > $_tmp_file
+    [ -s $_tmp_file ] || {
+        echo "is_sftp_enabled(): No sftp setting found" >> $logfile
+        rm -rf $_tmp_file
+        return 0
+    }
+    ret=$(grep -q "Enabled" $_tmp_file)
+    if [[ $ret -eq 0 ]]; then
+        echo "Enabled"
+    else
+        echo "Disabled"
+    fi
+}
+
+# Enable the sftp service on the controller.
+sftp_enable()
+{
+    _tmp_file="$tmpdir/sftp_enable"
+    [ -f $_tmp_file ] && rm -rf $_tmp_file
+    echo "sftp_enable(): Checking if sftp service is enabled" >> $logfile
+    is_sftp_enabled > $_tmp_file
+    grep -q "Enabled" $_tmp_file || {
```

Comment:
With changes above this can be changed to... 

`is_sftp_enabled || {`

No need to use temp_file. Same below.

Reply:
done.